src/display_buffer.cc

```diff
@@ -286,7 +286,7 @@ void DisplayBuffer::optimize()
         line.optimize();
 }
 
-DisplayLine parse_display_line(StringView line)
+DisplayLine parse_display_line(StringView line, HashMap<String, DisplayLine> atoms)
```

Comment:
Use a `const HashMap<...>&` here, as we only access the map for reading, so there is no reason to copy the hash map.